beetsplug/spotify.py

```diff
@@ -293,7 +293,13 @@ def album_for_id(self, album_id: str) -> AlbumInfo | None:
         if not (spotify_id := self._extract_id(album_id)):
             return None
 
-        album_data = self._handle_response("get", self.album_url + spotify_id)
+        try:
+            album_data = self._handle_response(
+                "get", self.album_url + spotify_id
+            )
+        except APIError:
+            return None
+
```

Comment:
This approach introduces a lot of duplication in the code. I think a cleaner approach would be to _never raise APIError_ in the first place.

See `_handle_response` method - I'd suggest removing all lines that raise `APIError` and returning `None` instead.